common/utxo.c

```diff
@@ -29,16 +34,22 @@ void towire_utxo(u8 **pptr, const struct utxo *utxo)
 struct utxo *fromwire_utxo(const tal_t *ctx, const u8 **ptr, size_t *max)
 {
 	struct utxo *utxo = tal(ctx, struct utxo);
+	u16 script_len;
+	u16 scriptsig_len;
 
 	fromwire_bitcoin_txid(ptr, max, &utxo->txid);
 	utxo->outnum = fromwire_u32(ptr, max);
 	utxo->amount = fromwire_amount_sat(ptr, max);
 	utxo->keyindex = fromwire_u32(ptr, max);
 	utxo->is_p2sh = fromwire_bool(ptr, max);
 
-	/* No need to tell hsmd about the scriptPubkey, it has all the info to
-	 * derive it from the rest. */
-	utxo->scriptPubkey = NULL;
+	script_len = fromwire_u16(ptr, max);
+	utxo->scriptPubkey = tal_arr(utxo, u8, script_len);
+	fromwire_u8_array(ptr, max, utxo->scriptPubkey, script_len);
```

Comment:
We now have `utxo->scriptPubkey = fromwire_tal_arrn(ptr, max, fromwire_u16(ptr, max));`